server/utils/queryHelpers/pubSanitize.js

```diff
@@ -15,11 +15,36 @@ const sanitizeHashes = (pubData, activePermissions) => {
 export default (pubData, initialData, releaseNumber) => {
 	const { loginData, scopeData } = initialData;
 	const { activePermissions } = scopeData;
-	const { canView, canViewDraft, canEdit, canEditDraft, canAdminCommunity } = activePermissions;
+	const { canView, canViewDraft, canEdit, canEditDraft } = activePermissions;
 
-	/* If there are no releases and the user does not have view access, they don't have access to the pub. */
-	/* Returning null will cause a 404 error to be returned. */
-	if (!pubData.releases.length && !canView && !canViewDraft) {
+	const hasPubMemberAccess = !!pubData.members.find((member) => {
+		return member.userId === initialData.loginData.id;
+	});
+	const visibleCollectionIds = initialData.communityData.collections.map((cl) => cl.id);
+	const filteredCollectionPubs = pubData.collectionPubs
+		? pubData.collectionPubs.filter((item) => {
+				return visibleCollectionIds.includes(item.collectionId);
+		  })
+		: [];
+	const hasCollectionMemberAccess = filteredCollectionPubs.reduce((prev, currCp) => {
+		const currCollection = initialData.communityData.collections.find((cl) => {
+			return currCp.collectionId === cl.id;
+		});
+		const hasCurrCollectionMemberAccess = !!currCollection.members.find((member) => {
+			return member.userId === initialData.loginData.id;
+		});
+		return prev || hasCurrCollectionMemberAccess;
+	}, false);
+	/* If there are no releases and the user does not have view access, they don't have scope-level */
+	/* We then must check if they have pub-level access and community-level access, otherwise */
+	/* we return null. Returning null will cause a 404 error to be returned. */
```

Comment:
Nit — `Returning null will cause a 404 error to be returned` might be more misleading than helpful; this code has several callers and it's up to them to decide what happens when `null` is returned.

Reply:
Good catch. Changed.